pkg/metrics/instance/manager.go

```diff
@@ -50,6 +50,9 @@ type Manager interface {
 	// one with Config.Name already exists.
 	ApplyConfig(Config) error
 
+	// ApplyConfigs batch applies the config
+	ApplyConfigs([]Config) (err error, successful []Config, failed []Config)
```

Comment:
This is a bit weird for a Go return signature. You could use custom error types here:

```go
type BatchApplyError struct {
   Failed map[Config]error 
   Successful []Config
}

func (e BatchApplyError) Error() string {
    return fmt.Sprintf("%d configs failed to apply", len(e.Failed)) 
} 
```

(Though I'm not sure if it really needs to return successful configs, since you could compute that if you really needed it)

Then you could use `errors.As` to unpack the error:

```go
var bae BatchApplyError 
if bae := BatchApplyError{}; errors.As(err, &bae) {
  // do something if needed 
}
```

Reply:
Cool! Created an object to hold the result, note the nonconfigerror isn't used now but will be once the configwatcher batch change goes in.